File: MEDIUM/227. Basic Calculator II/solution.py
class Solution:
    def calculate(self, s: str) -> int:
        n = len(s)
        sign = '+'
        num = 0
        stack = []
        for i in range(n):
            if s[i].isdigit():
                num = num*10+int(s[i])
            if (not s[i].isdigit() and s[i]!=' ') or i == n-1:
                if sign == '+':
                    stack.append(num)
                elif sign == '-':
                    stack.append(-num)
                elif sign == '/':
                    top = stack.pop()
                    stack.append(int(top/num))
                elif sign == '*':
                    top = stack.pop()
                    stack.append(num*top)
                num = 0
                sign = s[i]
        return sum(stack)

File: MEDIUM/227. Basic Calculator II/solution.cpp
class Solution {
public:
    int calculate(string s) {
       int n = s.size();
      stack<int>st;
        int num = 0;
    char sign = '+';
      for(int i = 0; i<n; i++){
        if(isdigit(s[i])){
            num = num*10+(s[i]-'0');
        }
        if((!isdigit(s[i]) and s[i]!=' ') or (i == n-1) ){
            if(sign == '+'){
                st.push(num);
            }
            else if(sign == '-'){
                st.push(-num);
            }
            else if(sign == '*'){
                int top = st.top();
                st.pop();
                st.push(top*num);
            }
            else if(sign == '/'){
                int top = st.top();
                st.pop();
                st.push(top/num);
            }
            num = 0;
            sign = s[i];
        }
      }
      int result = 0;
      while(!st.empty()){
        result+=st.top();
        st.pop();
      }
      return result;
      
    }
};
